libs/stokesian_dynamics/device_matrix.hpp

```diff
@@ -0,0 +1,658 @@
+// Copyright (c) 2018 Henri Menke. All rights reserved.
+
+#include <iomanip>
+#include <limits>
+#include <type_traits>
+
+#include "thrust_wrapper.hpp"
+
+#ifdef __CUDACC__
+#include <cublas_v2.h>
+#include <cusolverDn.h>
+#endif
+
+#ifdef __CUDACC__
+#define DEVICE_FUNC __host__ __device__
+#else
+#define DEVICE_FUNC
+#endif
+
+#ifdef __GNUC__
+#define MAYBE_UNUSED __attribute__((unused))
+#else
+#define MAYBE_UNUSED
+#endif
+
+namespace policy {
+
+struct host {
+    template <typename T>
+    using vector = thrust_wrapper::host_vector<T>;
+    // the remove_const is necessary, because things cannot be static and const
+    // at the same time
+    static std::remove_const<decltype(thrust_wrapper::host)>::type par()
+      { return thrust_wrapper::host; }
+};
+
+struct device {
+    template <typename T>
+    using vector = thrust_wrapper::device_vector<T>;
+    static std::remove_const<decltype(thrust_wrapper::device)>::type par()
+      { return thrust_wrapper::device; }
+};
+
+template <typename...>
+using void_t = void;
+
+template <typename Policy, typename = void>
+struct is_policy : std::false_type {};
+
+template <typename Policy>
+struct is_policy<Policy, void_t<typename Policy::template vector<class T>,
+                                decltype(Policy::par())>> : std::true_type {};
+
+} // namespace policy
+
+DEVICE_FUNC inline thrust_wrapper::tuple<std::size_t, std::size_t>
+unravel_index(std::size_t index, std::size_t lda) {
```

Comment:
Pls add a comment if this is row or col major

Reply:
in this particular place, it is not relevant, because this function is used only once for a symmetric matrix. However, I've added the information that it is col major to the class description.